lib/oli_web/plugs/authorize_section_preview.ex

```diff
@@ -0,0 +1,37 @@
+defmodule Oli.Plugs.AuthorizeSectionPreview do
+  import Plug.Conn
+  import Phoenix.Controller
+
+  alias Oli.Delivery.Sections
+
+  # We only allow access to preview mode if the user is logged in as an instructor
+  # enrolled in the course section. If the user is enrolled as a student,
+  # we redirect out of this mode to render the page in regular delivery mode.
+
+  def init(opts), do: opts
+
+  def call(conn, _opts) do
+    user = conn.assigns[:current_user]
+    section_slug = conn.path_params["section_slug"]
+
+    cond do
+      Sections.is_instructor?(user, section_slug) ->
+        conn
+      not is_nil(user) and Sections.is_enrolled?(user.id, section_slug) ->
+        redirect_path =
+          conn
+          |> current_path()
+          |> String.replace("preview/", "")
```

Comment:
This would be a problematic way to redirect away student attempts in the case where the section slug ends with `"preview"`. Consider the following example:

`"https://proton.oli.cmu.edu/sections/biology_preview/preview/page/intro"`



The `String.replace` would strip out a portion of the section slug.

Reply:
@darrensiegel, while @mcass19 is out, could you suggest a more ironclad search/replace pattern that would work, or should we look at a different strategy altogether?

Reply:
Yeah good catch @darrensiegel, wasn't completely sure if that would work for all cases.

Did the plug in this way to not have to write the redirect logic in each preview method. Will think if there's something safer we could do here or will re write it to be in each route specific.

Reply:
Yeah i think including the leading `/` is good enough.  That will avoid replacing any other component of the URL

Reply:
@darrensiegel the last changes were pushed, please take a look whenever you can.